### PR TITLE
separated config map resource into multiple files

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/resource/configmap/delete.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+	if len(configMapsToDelete) != 0 {
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "deleting the config maps in the Kubernetes API")
+
+		// Create the config maps in the Kubernetes API.
+		namespace := key.ClusterNamespace(customObject)
+		for _, configMap := range configMapsToDelete {
+			err := r.k8sClient.CoreV1().ConfigMaps(namespace).Delete(configMap.Name, &apismetav1.DeleteOptions{})
+			if apierrors.IsNotFound(err) {
+				// fall through
```
Comment (fgimenez) [inline service/resource/configmap/delete.go]: Same as above about logging the error

File: service/resource/configmap/create.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+	// Create the config maps in the Kubernetes API.
+	if len(configMapsToCreate) != 0 {
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "creating the config maps in the Kubernetes API")
+
+		namespace := key.ClusterNamespace(customObject)
+		for _, configMap := range configMapsToCreate {
+			_, err := r.k8sClient.CoreV1().ConfigMaps(namespace).Create(configMap)
+			if apierrors.IsAlreadyExists(err) {
+				// fall through
```
Comment (fgimenez) [inline service/resource/configmap/create.go]: Shouldn't this error be logged?

Comment (xh3b4sd) [reply service/resource/configmap/create.go]: Uncertain. This is actually just moving code. I would keep it as it is. I am undecided if this is something we should worry about. If we do not care about this special case we should just remove the extra branch of the condition.

Comment (xh3b4sd) [reply service/resource/configmap/create.go]: I merge this now. Let me know what you think. We can sort this asynchronously.

**Pull request merged**